NDCube 2.1.0 release broke Specutils docs build

### Describe the bug

Hi all,

 Since the NDCube 2.1.0 release, I have been unable to build the docs for `specutils`, running into this error (despite having the relevant NDCube class in our nitpick-exceptions file):

```
Extension error (numpydoc.numpydoc):
Handler <function mangle_signature at 0x7f4dd674f7e0> for event 'autodoc-process-signature' threw an exception (exception: Error parsing See Also entry '* `reproject.reproject_interp`' in the docstring of reproject_to in /home/docs/checkouts/readthedocs.org/user_builds/specutils/conda/999/lib/python3.11/site-packages/ndcube/ndcube.py.)
```

You can see the full traceback at https://readthedocs.org/projects/specutils/builds/19702847/. Any ideas on how I can avoid this, or is this something that needs to be remedied on your end? Thanks.

### To Reproduce

_No response_

### Screenshots

_No response_

### System Details

_No response_

### Installation method

_No response_

I still see the same error after adding the `reproject` intersphinx mapping:

```
[...]
loading intersphinx inventory from https://reproject.readthedocs.io/en/stable/objects.inv...
[...]
/Users/rosteen/opt/anaconda3/envs/specutils-dev/lib/python3.10/site-packages/numpydoc/docscrape.py:460: UserWarning: Unknown section Example in the docstring of axis_world_coords_values in /Users/rosteen/opt/anaconda3/envs/specutils-dev/lib/python3.10/site-packages/ndcube/utils/cube.py.
  warn(msg)
/Users/rosteen/opt/anaconda3/envs/specutils-dev/lib/python3.10/site-packages/numpydoc/docscrape.py:460: UserWarning: Unknown section Example in the docstring of crop in /Users/rosteen/opt/anaconda3/envs/specutils-dev/lib/python3.10/site-packages/ndcube/ndcube.py.
  warn(msg)
/Users/rosteen/opt/anaconda3/envs/specutils-dev/lib/python3.10/site-packages/numpydoc/docscrape.py:460: UserWarning: Unknown section Example in the docstring of crop_by_values in /Users/rosteen/opt/anaconda3/envs/specutils-dev/lib/python3.10/site-packages/ndcube/ndcube.py.
  warn(msg)

Extension error (numpydoc.numpydoc):
Handler <function mangle_signature at 0x7f93435b4ee0> for event 'autodoc-process-signature' threw an exception (exception: Error parsing See Also entry '* `reproject.reproject_interp`' in the docstring of reproject_to in /Users/rosteen/opt/anaconda3/envs/specutils-dev/lib/python3.10/site-packages/ndcube/ndcube.py.)
make: *** [html] Error 2
```

--- a/ndcube/ndcube.py
+++ b/ndcube/ndcube.py
@@ -124,8 +124,8 @@ class NDCubeABC(astropy.nddata.NDDataBase):
             their corresponding array dimensions, unless ``pixel_corners=True``
             in which case the length along each axis will be 1 greater than
             the number of pixels.
-        Example
-        -------
+        Examples
+        --------
         >>> NDCube.axis_world_coords('lat', 'lon') # doctest: +SKIP
         >>> NDCube.axis_world_coords(2) # doctest: +SKIP
 
@@ -171,8 +171,8 @@ class NDCubeABC(astropy.nddata.NDDataBase):
             their corresponding array dimensions, unless ``pixel_corners=True``
             in which case the length along each axis will be 1 greater than the number of pixels.
 
-        Example
-        -------
+        Examples
+        --------
         >>> NDCube.axis_world_coords_values('lat', 'lon') # doctest: +SKIP
         >>> NDCube.axis_world_coords_values(2) # doctest: +SKIP
 
@@ -214,8 +214,8 @@ class NDCubeABC(astropy.nddata.NDDataBase):
         -------
         result: `~ndcube..ndcube.NDCubeABC`
 
-        Example
-        -------
+        Examples
+        --------
         An example of cropping a region of interest on the Sun from a 3-D image-time cube:
         >>> point1 = [SkyCoord(-50*u.deg, -40*u.deg, frame=frames.HeliographicStonyhurst), None]  # doctest: +SKIP
         >>> point2 = [SkyCoord(0*u.deg, -6*u.deg, frame=frames.HeliographicStonyhurst), None]  # doctest: +SKIP
@@ -266,8 +266,8 @@ class NDCubeABC(astropy.nddata.NDDataBase):
         result: `~ndcube.ndcube.NDCubeABC`
 
 
-        Example
-        -------
+        Examples
+        --------
         An example of cropping a region of interest on the Sun from a 3-D image-time cube:
         >>> NDCube.crop_by_values((-600, -600, 0), (0, 0, 0), units=(u.arcsec, u.arcsec, u.s)) # doctest: +SKIP
 
@@ -710,9 +710,9 @@ class NDCubeBase(NDCubeABC, astropy.nddata.NDData, NDCubeSlicingMixin):
         See Also
         --------
 
-        * `reproject.reproject_interp`
-        * `reproject.reproject_adaptive`
-        * `reproject.reproject_exact`
+        reproject.reproject_interp
+        reproject.reproject_adaptive
+        reproject.reproject_exact
 
         Notes
         -----

